hutch_python/cli.py

```diff
@@ -109,6 +130,20 @@ def configure_ipython_session():
     ]
     ipy_config.InteractiveShellApp.exec_files = files
 
+    # Set up history from local disk (not NFS)
+    if args.hist_file is None:
+        hist_file = f"/u1/{os.environ['USER']}/hutch-python/history.sqlite"
```

Comment:
So the default is now to have each user write to their own history file?  What exactly are the conditions for having a `/u1/<user>` directory on a given host?  I don't have one on psbuild so this defaults to the standard location (~/.ipython/profile_default/...)

Reply:
As written this _only_ affects e.g. xppopr running on xpp-daq and similar operator accounts.
If you run as zlentz for example, you'll use the normal per-user file.

Reply:
Would it be prudent for normal users to have a fallback to some per-host NFS location in the ipython config directory?

Reply:
I think so, in the interest of keeping original ipython functionality available.  I'd personally have left that as the default, and made the flag generate the file on a local disk.  

Is it possible to do:
- no-flag: normal NFS location
- flag, no filename: default hist_file name in `/u1/...`
- flag, filename: history at the provided filename?

Reply:
I'll look into the flexibility of argparse tomorrow, but I agree that I should have made the default behavior no change from normal.

I'd like it to "just work" without too much bash-fu in e.g. the xcs3 launcher script (e.g. if xppopr do this, if not do that... is very annoying).

Reply:
I imagine adding just switching the xpppython script to read:
```
hutch-python --cfg "${HERE}/conf.yml" --hist-file
```
isn't too invasive, and we can switch it on as we get complaints about it.

Reply:
I think `nargs='?'` might work here:

```python
parser = argparse.ArgumentParser()
parser.add_argument('--hist-file', dest='hist_file', nargs='?', const='/u1/...')
parser.parse_args([])  # Namespace(hist_file=None)
parser.parse_args(['--hist-file'])  # Namespace(hist_file='/u1/...')
parser.parse_args(['--hist-file', 'my_file/path'])  # Namespace(hist_file='my_file/path')
```

Reply:
I'll implement it exactly as you suggest, it's the cleanest overall and is opt-in

Reply:
Implemented as suggested, edited a bit so I could include another test, expanded to allow any environment vars, not just user, because why not